chart/templates/deployment.yaml

```
@@ -54,3 +54,15 @@ spec:
         - name: config
           configMap:
             name: {{ .Release.Name }}-config
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```

Why not use this directly? Same applies to below.

```suggestion
      nodeSelector:
        {{- toYaml .Values.nodeSelector | nindent 8 }}
```

---

Because it will leave empty `nodeSelector:` (and others below) if no value is specified. With `with` you change the scope but it also behaves as `if` and it only executes if `.Values.nodeSelector` is not empty.

---

Ah nice, didn't know!